tools/cache.py

```diff
@@ -46,14 +46,17 @@ def get(self, shortname, creator, extension='.bc', what=None):
 
 # Given a set of functions of form (ident, text), and a preferred chunk size,
 # generates a set of chunks for parallel processing and caching.
-def chunkify(funcs, chunk_size, DEBUG=False):
+def chunkify(funcs, chunk_size, DEBUG=False, using_fn_implementations=False):
```

Comment:
While I have not been able to repro the original issue, I went ahead and moved the version of splitting funcs that doesn't rely on function boundaries to match '\n}\n' into split_funcs. This removed the need to modify cache.chunkify. This should do the job.

    def split_funcs(js, just_split=False):
      if just_split: return map(lambda line: ('(json)', line), js.split('\n'))

      funcs = []
      prev_match = None

      for match in func_sig.finditer(js):
        if prev_match is not None:
          ident = prev_match.group(1)
          func = js[prev_match.start(0):match.start(0)]
          funcs.append((ident, func))
        prev_match = match

      if prev_match is not None:
          ident = prev_match.group(1)
          func = js[prev_match.start(0):]
          funcs.append((ident, func))

      return funcs